### PR TITLE
fix(bake/manifests): delete templates when request finishes

### PR DESCRIPTION
**Pull request by lwander**
https://github.com/spinnaker/spinnaker/issues/2636

### CODE REVIEW

File: rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java [hunk 1]
```diff
@@ -47,4 +52,39 @@ public Path downloadArtifactToTmpFile(Artifact artifact) throws IOException {
 
     return path;
   }
+
+  public static class BakeManifestEnvironment {
+    @Getter
+    final private Path stagingPath = Paths.get(System.getProperty("java.io.tmpdir"), UUID.randomUUID().toString());
+
+    public BakeManifestEnvironment() {
+      stagingPath.toFile().mkdirs();
```
Comment (duftler) [inline rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java]: Might be worth a `log.debug` here.

File: rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java [hunk 2]
```diff
@@ -47,4 +52,39 @@ public Path downloadArtifactToTmpFile(Artifact artifact) throws IOException {
...
+    }
+
+    private boolean recursiveDeleteFile(File file) {
```
Comment (duftler) [inline rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java]: Can use `FileSystemUtils.deleteRecursively()`.

File: rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java (hunk 1 above)
Comment (lwander) [reply rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java]: :+1:

File: rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java (hunk 2 above)
Comment (lwander) [reply rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java]: Wasn't imported yet -- will do

File: rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java (hunk 1 above)
Comment (lwander) [reply rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java]: Upgraded to `warn` on failure

Comment (duftler) [reply rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/TemplateUtils.java]: My thinking was if it doesn't fail, but isn't obvious where it was created (or isn't where one expected), they could use debug logging to troubleshoot.

**Pull request merged**